tests/publiccloud/patch_and_reboot.pm

```diff
@@ -45,12 +45,14 @@ sub run {
     record_info('UNAME', $args->{my_instance}->ssh_script_output(cmd => 'uname -a'));
     $args->{my_instance}->ssh_assert_script_run(cmd => 'rpm -qa > /tmp/rpm-qa.txt');
     $args->{my_instance}->upload_log('/tmp/rpm-qa.txt');
-    $args->{my_instance}->cleanup_cloudinit() if (is_cloudinit_supported);
-    $args->{my_instance}->softreboot(timeout => get_var('PUBLIC_CLOUD_REBOOT_TIMEOUT', 600));
 
     if (is_cloudinit_supported) {
+        $args->{my_instance}->cleanup_cloudinit();
+        $args->{my_instance}->softreboot(timeout => get_var('PUBLIC_CLOUD_REBOOT_TIMEOUT', 600), scan_ssh_host_key => 1);
```

Comment:
Why `reboot` moved here? why now is done only for cloudinit?

Reply:
There is also `else` block. The difference is that if we do `cloud-init clean` we need to re-scan the ssh host key.